RPGManager_development/test/unit_test/character_sheet_unit_tests/ArmorsUnitTests.java

```diff
@@ -0,0 +1,142 @@
+
+import rpg_database.character_sheet.Armors;
+import rpg_database.character_sheet.Background;
+import rpg_database.character_sheet.BaseClasses;
+import rpg_database.character_sheet.CharacterSheet;
+import rpg_database.character_sheet.Fields;
+
+public class ArmorsUnitTests {
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Test
+	public void expectException_SetArmorsMalformedInput() {
+		expectExceptionWithMessage(InvalidParameterException.class,
+				"class java.lang.String value is not an instance of class rpg_database.character_sheet.Armors");
+		final String malformedInput = "MALFORMED INPUT";
+		CharacterSheet characterSheet = new CharacterSheet("CharacterSheet");
+		characterSheet.setData(Fields.ARMOR_TYPE, malformedInput);
+	}
+
+	private void expectExceptionWithMessage(Class<? extends Exception> exceptionClass, String message) {
+		thrown.expect(exceptionClass);
+		thrown.expectMessage(message);
+	}
+
+	@Test
+	public void testSetArmorTypeLightPlate() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.ARMOR_TYPE, Armors.LIGHT_PLATE);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.LIGHT_PLATE, actualValue);
+	}
+
+	@Test
+	public void testGetDefaultArmorForWarrior() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.WARRIOR);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.LIGHT_MAIL, actualValue);
+	}
+
+	@Test
+	public void testGetDefaultArmorForRogue() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.ROGUE);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.LIGHT_LEATHER, actualValue);
+	}
+
+	@Test
+	public void testGetDefaultArmorForMage() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.MAGE);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.NONE, actualValue);
+	}
+
+	@Test
+	public void testSetArmorForWarrior() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.WARRIOR);
+		characterSheet.setData(Fields.ARMOR_TYPE, Armors.HEAVY_MAIL);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.HEAVY_MAIL, actualValue);
+	}
+
+	@Test
+	public void testGetDexterityArmorForWarrior() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.ARMOR_TYPE, Armors.HEAVY_MAIL);
+		int actualValue = characterSheet.getData(Fields.DEXTERITY_VALUE);
+		assertEquals(-3, actualValue + Armors.class.cast(characterSheet.getData(Fields.ARMOR_TYPE)).getArmorPenalty());

```

Comment:
- ```Armors.class.cast(characterSheet.getData(Fields.ARMOR_TYPE)).getArmorPenalty()``` helyett:
```java
characterSheet.<Armors>getData(Fields.ARMOR_TYPE)).getArmorPenalty()
```
és miért kell hozzáadnod a dexterity-hez? Nem elég simán a penalty értékét vizsgálni? A DEX itt amúgy is 0 nem?
- Az armor training ezt is befolyásolja, ugyanis a penalty-t csak speedre kapja a karakter ha van neki. Ez is átírásszagú a jövőben, érdemes rá taskot felvenni, vagy jelezni az előzőben ha azt már felvetted.